cpp/src/binaryop/binaryop.cpp

```diff
@@ -229,26 +189,33 @@ void binary_operation(mutable_column_view& out,
                       const std::string& ptx,
                       rmm::cuda_stream_view stream)
 {
+  jitify2::ProgramCache<> binaryop_program_cache(
+    /*max_size = */ 100, *binaryop_jit_kernel_cu_jit);
+
   std::string const output_type_name = cudf::jit::get_type_name(out.type());
 
   std::string ptx_hash =
-    hash + "." + std::to_string(std::hash<std::string>{}(ptx + output_type_name));
+    "prog_binop." + std::to_string(std::hash<std::string>{}(ptx + output_type_name));
   std::string cuda_source =
     "\n#include <cudf/types.hpp>\n" +
```

Comment:
Can you try to remove this and see if it works this time.

Reply:
ah, good catch, yes it should work.

Reply:
I had to move include this in to `operation.hpp` prior to including `operation-udf.hpp` (which gets replaced at runtime).